projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3363,12 +3363,26 @@ export abstract class IgxGridBaseDirective implements GridType,
     private get hasZeroResultFilter(): boolean {
         return this.filteredData && this.filteredData.length === 0;
     }
+    protected get totalCalcWidth() {
+        return this.platform.isBrowser ? this.calcWidth : undefined;
+    }
+
+    protected get renderData() {
+        // omit data if not in the browser and size is %
+        return !this.platform.isBrowser && this.isPercentHeight ? undefined : this.data;
+    }
+
+    @HostBinding('style.display')
```

Comment:
Shouldn't those host bindings be in the scss in the themes

Reply:
They already are (you can find the same styles in the theme), but they don't get applied until the theme is fully loaded and they are necessary for the body content to stretch correctly in SSR. Hence why they are set inline as well.